### PR TITLE
generate-docs target added to CMake. It will preprocess the readme.

### PR DESCRIPTION
**Pull request by OzanCansel**
- `generate-docs` target added to CMake.
- It will preprocess the readme.md and generates the doxygen docs.

It basically converts the relative links to absolute urls.

### CODE REVIEW

**Review by ihsandemir — CHANGES_REQUESTED** [review]
The following does not work in MacOS:

`scripts/generate-docs.sh README.md 5.1  "https://github.com/hazelcast/hazelcast-cpp-client"`

File: CMakeLists.txt [hunk 1]
```diff
@@ -251,3 +251,24 @@ message(STATUS "  DISABLE_LOGGING =       ${DISABLE_LOGGING}")
 message(STATUS "  CMAKE_INSTALL_PREFIX =  ${CMAKE_INSTALL_PREFIX}")
 message(STATUS "  BUILD_TESTS =           ${BUILD_TESTS}")
 message(STATUS "  BUILD_EXAMPLES =        ${BUILD_EXAMPLES}")
+
+if ( UNIX )
```
[the remark this answers is not in the document]
Comment (OzanCansel) [reply CMakeLists.txt:255]: I only targeted linux systems and didn't think that it needs to be a cross-platform script(s). It is easier to port to `MacOS` but do we really need to port to `Windows` ?

File: scripts/generate-docs.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+MAINPAGE=${1}
```
[the remark this answers is not in the document]
Comment (OzanCansel) [reply scripts/generate-docs.sh:3]: Let me add

**Comment by OzanCansel** [comment]
Ported to MacOS. `grep` and `doxygen` are needed to be installed.

```sh
brew install grep doxygen
```

**Pull request closed without merging**